core/services/relay/evm/bindings.go

```diff
@@ -0,0 +1,50 @@
+package evm
```

Comment:
I feel like bindings are too complex for no reason, which makes chain reader harder to understand. I made a PR that goes into this one that should clear things up a bit https://github.com/smartcontractkit/chainlink/pull/11561/files

Reply:
Left some comments, but the main thing is that if it's a method or event should be determined by the ABI not, the caller.  It's more flexible and easier to verify.

I do like the idea of encapsulating it more, I can take another look soon.

Reply:
I reshuffled things in a couple of different ways and wasn't fully satisfied with any of those. I will spend a bit more time on it later and do it as a separate PR that isn't connected to this